### PR TITLE
doc: add dash doc feed & fix rustdoc warnings

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Also fix all doc warnings, and reject warnings in CI in the future

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
All good except deployment failure

image

File: .github/workflows/doc.yml [hunk 1]
```diff
@@ -18,24 +23,57 @@ jobs:
...
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-docset
+      - name: build docsets
+        run: |
+          cargo docset --no-clean
+          tar -cvzf risingwave.docset.tgz target/docset/risingwave.docset
+
+          echo "<entry>
+          <version>$(date -u +%Y-%m-%dT%H:%M:%SZ)</version>
```
Comment (xxchan) [inline .github/workflows/doc.yml:63]: https://kapeli.com/docsets#dashdocsetfeed

> You can use any versioning system you want. Dash will use string comparison to determine whether or not to download an update.

So I think timestamp might be good

**Review by wangrunji0408 — APPROVED** [review]
Amazing.

File: src/connector/src/source/data_gen_util.rs [hunk 2]
```diff
@@ -18,9 +18,12 @@ use futures::{pin_mut, Stream, StreamExt};
 use tokio::runtime::Runtime;
 use tokio::sync::mpsc;
 
-/// Spawn the data generator to a dedicated runtime, returns a channel receiver
-/// for acquiring the generated data. This is used for the [`DatagenSplitReader`] and
-/// [`NexmarkSplitReader`] in case that they are CPU intensive and may block the streaming actors.
+/// for acquiring the generated data. This is used for the [`DatagenSplitReader`]
```
Comment (xxchan) [inline src/connector/src/source/data_gen_util.rs:21]:
```suggestion
/// Spawn the data generator to a dedicated runtime, returns a channel receiver
/// for acquiring the generated data. This is used for the [`DatagenSplitReader`]
```

**Comment by xxchan** [comment]
Testing on my fork:
https://github.com/xxchan/risingwave/actions/workflows/doc.yml
https://xxchan.me/risingwave/

**Pull request merged**